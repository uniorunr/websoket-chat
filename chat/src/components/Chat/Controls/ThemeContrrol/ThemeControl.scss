@import "../../../../theme/colors";

.theme-control {
  display: flex;
  align-items: center;
  flex-grow: 1;

  .theme-control__button-mobile {
    display: none;
    background: none;
    border: none;
    outline: none;
    cursor: pointer;
  }

  .theme-control__button {
    background: $black-color;
    color: $white-color;
    text-transform: uppercase;
    padding: 5px 15px;
    outline: none;
    cursor: pointer;
    border: 2px solid $black-color;
    border-radius: 5px;
  }

  .theme-control__button:hover {
    font-weight: 500;
    background: $white-color;
    color: $black-color;
    border-radius: 5px;
  }
}

@media only screen and (max-width : 549px) {
  .theme-control {
    .theme-control__button-mobile {
      display: flex;
    }

    .theme-control__button {
      display: none;
    }
  }
}
