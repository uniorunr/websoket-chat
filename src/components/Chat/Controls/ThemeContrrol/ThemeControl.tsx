import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import * as actions from '../../../../actions/actions';
import ThemeMobileIcon from '../../../../assets/icons/rainbow.png';
import './ThemeControl.scss';

const ThemeControl = ({ showSettingsModalWindow }: ThemeControlProps) => {
  const handleClick = () => {
    showSettingsModalWindow(true);
  };

  return (
    <div className="theme-control">
      <button
        type="button"
        className="theme-control__button-mobile"
        onClick={handleClick}
      >
        <img src={ThemeMobileIcon} alt="Theme choose icon" width={25} />
      </button>
      <button
        type="button"
        className="theme-control__button"
        onClick={handleClick}
      >
        settings
      </button>
    </div>
  );
};

interface ThemeControlProps {
  showSettingsModalWindow: Function;
}

const mapDispatchToProps = (dispatch: Dispatch) => {
  const { showSettingsModalWindow } = bindActionCreators(actions, dispatch);
  return {
    showSettingsModalWindow: (status: boolean) => {
      showSettingsModalWindow(status);
    }
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(ThemeControl);
