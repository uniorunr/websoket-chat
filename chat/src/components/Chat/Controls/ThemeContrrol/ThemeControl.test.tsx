import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../../store/mainStore';
import ThemeControl from './ThemeControl';

it('ThemeControl component renders correctly', () => {
  const app = shallow(
      <Provider store={store}>
        <ThemeControl />
      </Provider>
  );
  expect(app).toMatchSnapshot();
});
