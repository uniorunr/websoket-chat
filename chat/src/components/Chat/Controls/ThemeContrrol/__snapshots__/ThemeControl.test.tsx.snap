// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeControl component renders correctly 1`] = `ShallowWrapper {}`;
